reconcile/terraform_tgw_attachments.py

```diff
@@ -1,157 +1,215 @@
 
-@defer
-def run(
-    dry_run, print_to_file=None, enable_deletion=False, thread_pool_size=10, defer=None
+    account = _account_with_assume_role_data(
+        peer_connection, cluster_name, cluster_region, cluster_cidr_block, ocm
+    )
+
+    # accepter is the cluster's AWS account
+    accepter = _build_accepter(
+        peer_connection,
+        account,
+        cluster_region,
+        cluster_cidr_block,
+        awsapi,
+    )
+    if accepter["vpc_id"] is None:
+        logging.error(f"[{cluster_name}] could not find VPC ID for cluster")
+        yield None
+
+    account_tgws = awsapi.get_tgws_details(
+        account,
+        cluster_region,
+        cluster_cidr_block,
+        tags=json.loads(peer_connection.get("tags") or "{}"),
+        route_tables=peer_connection.get("manageRoutes"),
+        security_groups=peer_connection.get("manageSecurityGroups"),
+        route53_associations=peer_connection.get("manageRoute53Associations"),
+    )
+    for tgw in account_tgws:
+        connection_name = f"{peer_connection['name']}_{account['name']}-{tgw['tgw_id']}"
+        requester = _build_requester(peer_connection, account, tgw)
+        item = {
+            "connection_provider": TGW_CONNECTION_PROVIDER,
+            "connection_name": connection_name,
+            "requester": requester,
+            "accepter": accepter,
+            "deleted": peer_connection.get("delete", False),
+        }
+        yield item
+
+
+def _account_with_assume_role_data(
+    peer_connection: dict,
+    cluster_name: str,
+    region: str,
+    cidr_block: str,
+    ocm: OCM,
+) -> dict[str, Any]:
+    account = peer_connection["account"]
+    # assume_role is the role to assume to provision the
+    # peering connection request, through the accepter AWS account.
+    provided_assume_role = peer_connection.get("assumeRole")
+    # if an assume_role is provided, it means we don't need
+    # to get the information from OCM. it likely means that
+    # there is no OCM at all.
+    account["assume_role"] = (
+        provided_assume_role
+        if provided_assume_role
+        else ocm.get_aws_infrastructure_access_terraform_assume_role(
+            cluster_name, account["uid"], account["terraformUsername"]
+        )
+    )
+    account["assume_region"] = region
+    account["assume_cidr"] = cidr_block
+    return account
+
+
+def _build_accepter(
+    peer_connection: dict,
+    account: dict,
+    region: str,
+    cidr_block: str,
+    awsapi: AWSApi,
+) -> dict:
+    (vpc_id, route_table_ids, subnets_id_az) = awsapi.get_cluster_vpc_details(
+        account,
+        route_tables=peer_connection.get("manageRoutes"),
+        subnets=True,
+    )
+    return {
+        "cidr_block": cidr_block,
+        "region": region,
+        "vpc_id": vpc_id,
+        "route_table_ids": route_table_ids,
+        "subnets_id_az": subnets_id_az,
+        "account": account,
+    }
+
+
+def _build_requester(
+    peer_connection: dict,
```

Comment:
Our convention is to use abstract collection types for function args if possible https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-style-guide/#abstract-collection-types

`_build_requester(peer_connection: Mapping ...) -> dict`

Reply:
updated